// jsunit/lib/Action.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Action = function () {
    function Action(_ref) {
        var _this = this;

        var context = _ref.context;
        var payload = _ref.payload;

        _classCallCheck(this, Action);

        var promise = undefined;

        this.context = context;

        promise = new Promise(function (resolve, reject) {
            return _this.execute({
                payload: payload,
                resolve: resolve,
                reject: reject,
                context: context
            });
        });

        return promise.catch(function (error) {
            throw new Error(error);
        });
    }

    _createClass(Action, [{
        key: 'execute',
        value: function execute() {
            throw new Error('Default execute not implemented');
        }
    }]);

    return Action;
}();

exports.default = Action;
